### PR TITLE
fix(groups): do a left join for group props

### PR DESCRIPTION
**Pull request by yakkomajuri**
## Problem

We currently do an inner join for group properties, meaning that if a filter is set for a group property like (pseudocode): 

`group1.properties.isDemo ≠ true` we will only return events that have an associated group but then don't have that property / have that property set to another value.

## Changes

I'd rather like to argue that the expected behavior is also to match events without a group at all. 

As such, we need to do a LEFT JOIN instead of an INNER JOIN. 

Tests seem to pass locally (let's see with the full suite here), but unsure if there were other considerations when this was made into an INNER JOIN.

## How did you test this code?

Added a test, expecting all others will pass as well.

### CODE REVIEW

File: posthog/queries/test/test_trends.py [hunk 1]
```diff
@@ -6317,6 +6317,57 @@ def test_filtering_with_group_props(self):
 
         response = Trends().run(filter, self.team)
         self.assertEqual(response[0]["count"], 1)
+        self.assertEqual(1, 2)
```
Comment (yakkomajuri) [inline posthog/queries/test/test_trends.py:6320]:
```suggestion
```

**Comment by yakkomajuri** [comment]
Over the last month only 5 teams ran queries with "negative" group property filters, for a total of 35 unique queries. These queries will see their results change, but there are so few that that's ok.

A good chunk of unique queries is me and the team that reported the issue this solves in the first place too.

Screenshot 2023-05-08 at 11 56 44

**Review by mariusandra — APPROVED** [review]
Looks more correct to me.

**Comment by yakkomajuri** [comment]
Closes #15338

**Pull request merged**